OPTIONAL_INSTALL.md

```diff
@@ -1,17 +1,250 @@
+# Detailed installation guide
```

Comment:
Perhaps we can link to this file directly from the readme?

Reply:
and we should change the name of the file

Reply:
I changed the name and linked to the detailed install instruction in the readme.